src/main/java/com/conveyal/osmlib/OSM.java

```diff
@@ -181,6 +181,7 @@ public void readFromFile(String filePath) {
                 // FIXME this takes two minutes on NL OSM. We should probably save the intersections in a MapDB table.
                 LOG.info("Detecting intersections...");
                 for (Way way : ways.values()) {
+                    if (way.hasTag("building")) continue;
```

Comment:
Yes, that's not a bad idea but it makes the library less general-purpose; we could add a flag to do that as part of a separate pull request maybe.